### ISSUE TITLE
Trainer not logging into Tensorboard

### ISSUE THREAD

**Issue by thomas-happify** [huggingface/transformers#11039]
## Environment info

- `transformers` version: 4.5.0.dev0
- Platform: Ubuntu 18.04.5 LTS (x86_64)
- Python version: 3.7.0
- PyTorch version (GPU?): 1.7.1+cu101
- Tensorflow version (GPU?):
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: Yes

### Who can help
@sgugger

## Information

Model I am using (Bert, XLNet ...): EncoderDecoderModel

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

This is the tensorboard logs 
https://tensorboard.dev/experiment/caY7XIGGTbK2Zfr2DTeoJA/#scalars
1. go to the `Text` tab [here](https://tensorboard.dev/experiment/caY7XIGGTbK2Zfr2DTeoJA/#text), you can see that `"logging_first_step": true, "logging_steps": 2`
2. `epoch` graph is showing 75 total steps, but no scalars were logged except for the first_step
<img width="413" alt="Screen Shot 2021-04-02 at 10 14 49 AM" src="https://user-images.githubusercontent.com/66082334/113423326-48c54080-939c-11eb-9c61-b8fde0d62d12.png">

```
[INFO|trainer.py:402] 2021-04-02 10:05:50,085 >> Using amp fp16 backend
[INFO|trainer.py:1013] 2021-04-02 10:05:50,181 >> ***** Running training *****
[INFO|trainer.py:1014] 2021-04-02 10:05:50,182 >>   Num examples = 100
[INFO|trainer.py:1015] 2021-04-02 10:05:50,182 >>   Num Epochs = 3
[INFO|trainer.py:1016] 2021-04-02 10:05:50,182 >>   Instantaneous batch size per device = 1
[INFO|trainer.py:1017] 2021-04-02 10:05:50,182 >>   Total train batch size (w. parallel, distributed & accumulation) = 4
[INFO|trainer.py:1018] 2021-04-02 10:05:50,182 >>   Gradient Accumulation steps = 1
[INFO|trainer.py:1019] 2021-04-02 10:05:50,182 >>   Total optimization steps = 75
{'loss': 13.7546, 'learning_rate': 2.5000000000000002e-08, 'epoch': 0.04}
100%|█████████████████████████████████████████████████████████████████████████████████| 75/75 [01:03<00:00,  1.28it/s][INFO|trainer.py:1196] 2021-04-02 10:06:53,931 >>

Training completed. Do not forget to share your model on huggingface.co/models =)


{'train_runtime': 63.7497, 'train_samples_per_second': 1.176, 'epoch': 3.0}
100%|█████████████████████████████████████████████████████████████████████████████████| 75/75 [01:03<00:00,  1.18it/s]
[INFO|trainer.py:1648] 2021-04-02 10:06:54,265 >> Saving model checkpoint to ./pretrain_decoder/
```

## Expected behavior
It should log training loss very other `logging_steps` right? or did I misunderstood?

**Comment by sgugger** [comment]
Could you check if the bug persists with a [source install](https://huggingface.co/transformers/installation.html#installing-from-source)? I'm trying to reproduce but I have proper logs on my side.

**Comment by thomas-happify** [comment]
@sgugger It only happened when I use `torch.distributed.launch`.
What might cause this problem?

**Comment by sgugger** [comment]
I'm still not able to reproduce: I see logs for each logging step in a distributed training as well.

**Comment by thomas-happify** [comment]
@sgugger  Ok.. probably only happens on my end.
Thanks
